referencing/what.md

```diff
@@ -0,0 +1,61 @@
+| 3.x         | Link Object   | `"operationRef"` | allowed, except `"operationId"` | Treat the reference target as the target of the link described by the Link Object |
+| 3.x         | Descriminator Object | `"mapping"` | n/a | for each name under `"mapping"`, if the value is not the name of a Schema Object under the Components Object, treat it as a reference to the schema to use when the discriminator field matches the mapping name |
+
+Note that not all references are handled by replacing the source with the target.  Some are defined in terms of behavior, and **cannot** be replaced with an inline value.
+
+### URIs, URLs, and URI-references
+
+All of the above keywords take a *[URI-reference](https://www.rfc-editor.org/rfc/rfc3986.html#section-4.1)* as a value, which can be either a *[URI](https://www.rfc-editor.org/rfc/rfc3986.html#section-3)* (which MUST include a [scheme](https://www.rfc-editor.org/rfc/rfc3986.html#section-3.1)) or a *[relative reference](https://www.rfc-editor.org/rfc/rfc3986.html#section-4.2)*, which must be resolved against a [base URI](https://www.rfc-editor.org/rfc/rfc3986.html#section-5.1) to produce a full URI.  In this guide, the term ***URI-reference resolution*** is always used for the process of resolving a URI-reference against a base URI.  
+
+Somewhat confusingly, the OAS uses the the terms "URI" and "URL" to mean "URI or URI-reference" and "URL or URL-reference", respectively.  In this guide, whenever a reference's "URI" is mentioned, it is always refers to the proper URI after any necessary URI-reference resolution has been performed.
+
+Also, in this guide the term "reference" alone *always* means reference as defined in the table above, and the words "resolve", "resolved", and "resolution" on their own *always* refer to the OAS-defined process that occurs *after* URI-reference resolution.
+
+## Non-reference linkages
+
+The following connections between parts of the OAS require that some name or identifier has been seen.  Due to ambiguities in the specification (which the OAI will hopefully be able to clarify in future patch releases), it is not always clear how to handle identifiers that are present in a _document_ that has been referenced, but are not present in the _exact JSON object_ targeted by any single reference.
+
+Note that some of the identifiers are not in objects that can be referenced, but are included for completeness.
+
+| OAS Version | Identifying Object | Identifying Field | Object of Identifier Use |
+| ----------- | ------------------ | ----------------- | ------------------------ |
+| 3.x         | Components Object  | Schema name       | Discriminator Object     |
+| 3.x         | Components Object  | Security Scheme name | Security Requirement Object |
+| 3.x         | Tag Object         | `name`            | Operation Object |
+| 3.x         | Operation Object   | `operationId`     | Link Object |
+| 3.x         | Server Object      | n/a               | Paths Object |
```

Comment:
Actually, in my notes Parameter Object and Server Variable Object are included as linkages.  I left them out because they felt different, mostly because they connect things that are close together.  But come to think of it, the Parameters Object can go in two places, and be `$ref`'d, so it's probably good to include.  And might as well include the Server Variable Object even if it is more localized.  There's also the `parameters` field in the Link Object which I'm not sure I noticed before now.  I'll add all of these.

Feel free to go hunting!  There might be more hiding somewhere, it's a big spec :-)